src/components/walletConnect.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import useWalletAccount from '../utils/persistAccount';
+import AccountButton from './accountButton';
+import DisconnectWalletButton from './disconnectWallet';
+import WalletConnectModal from './walletConnectModal';
+
+const WalletConnect = () => {
+  const [account, setAccount] = useWalletAccount();
+  if (!account) {
+    return <WalletConnectModal />
+  }
+  return (
+    <>
+      <AccountButton address={account.address} />
+      <DisconnectWalletButton />
+    </>
+  )
+}
+
+export default WalletConnect;
```

Comment:
Wallet connect is like dispatch component, it sits in navbar and if no accounts are connected, show the button that says "Connect Wallet", if an account is connected if show a button that has the truncated address name and the disconnect button.